modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase

```diff
@@ -237,6 +237,138 @@ definition {
 		}
 	}
 
+	@description = "LPS-146871 - Verify that it's possible to create an Action with Groovy Script"
+	@priority = "5"
+	test CanCreateActionWithGroovyScript {
+		property portal.acceptance = "true";
+
+		task ("Given a Groovy Script") {
+			var groovyScript = '''
+			import com.liferay.object.service.ObjectEntryLocalServiceUtil;
+			import com.liferay.object.model.ObjectEntry;
+			import com.liferay.portal.kernel.service.ServiceContext;
+
+			import java.io.Serializable;
+			import java.util.Map;
+
+			ObjectEntry objectEntry = ObjectEntryLocalServiceUtil.getObjectEntry(objectEntryId);
+
+			Map<String, Serializable> values = objectEntry.getValues();
+
+			values.put("customObjectFieldActionTest", "Action Test Works")
+
+			System.out.println(objectEntry);
+
+			ObjectEntryLocalServiceUtil.updateObjectEntry(objectEntry.getUserId(), objectEntryId, values, new ServiceContext());
+			''';
+		}
+
+		task ("Given an object two fields is created") {
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object",
+				objectName = "CustomObject",
+				pluralLabelName = "Custom Objects");
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Text",
+				fieldLabelName = "Custom Field",
+				fieldName = "customObjectField",
+				fieldType = "String",
+				isRequired = "true",
+				objectName = "CustomObject");
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Text",
+				fieldLabelName = "Custom Field Action Test",
+				fieldName = "customObjectFieldActionTest",
+				fieldType = "String",
+				isRequired = "false",
+				objectName = "CustomObject");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject");
+		}
```

Comment:
We should only have one given here. We can combine this into one task: `Given a Grooy Script and an object with two text fields are created`